File: 2116-count-number-of-pairs-with-absolute-difference-k/count-number-of-pairs-with-absolute-difference-k.kt
class Solution {
    fun countKDifference(nums: IntArray, k: Int): Int {
        val map = mutableMapOf<Int, Int>()
        var count = 0
        for(num in nums) {
            map[num] = map.getOrDefault(num, 0) + 1
            count += map.getOrDefault(num - k, 0) + map.getOrDefault(num + k, 0)
        }
        return count
    }
}

File: 2116-count-number-of-pairs-with-absolute-difference-k/count-number-of-pairs-with-absolute-difference-k.swift
class Solution {
    func countKDifference(_ nums: [Int], _ k: Int) -> Int {
        var containIndex = [[Int]]()

        var count = 0
        nums.forEach {
            count += 1
            for j in nums.enumerated() {
                if count - 1 != j.offset {
                        if j.element == abs($0 + k) || j.element == $0 - k {
                        var temp = [count - 1, j.offset].sorted()
                        containIndex.append(temp)
                    }
                }
            }
        }

        print(containIndex)
        
        return Set(containIndex).count
    }
}